torch/autograd/function.py

```diff
@@ -38,24 +39,9 @@ def mark_dirty(self, *args):
         self.dirty_tensors = args
 
     def mark_shared_storage(self, *pairs):
-        """Marks that given pairs of distinct tensors are sharing storage.
-
-        **This should be called at most once, only from inside the**
-        :func:`forward` **method, and all arguments should be pairs of
-        (input, output).**
-
-        If some of the outputs are going to be tensors sharing storage with
-        some of the inputs, all pairs of (input_arg, output_arg) should be
-        given to this function, to ensure correctness checking of in-place
-        modification. The only exception is when an output is exactly the same
-        tensor as input (e.g. in-place ops). In such case it's easy to conclude
-        that they're sharing data, so we don't require specifying such
-        dependencies.
-
-        This function is not needed in most functions. It's primarily used in
-        indexing and transpose ops.
-        """
-        self.shared_pairs = pairs
+        warnings.warn(
```

Comment:
I just realized that we won't be able to track sharing if someone uses `set_` (which is not an error on Variables that don't require grad, right), correct? We might want to keep it after all

Reply:
That's true. I don't think `mark_shared_storage` is a good solution though. I think we should be working towards an API that matches the underlying semantics instead of this sort of inversion of control.

In this case, that would mean exposing a `Variable` constructor that takes a `base`/`view` or fixing `set_`.

Reply:
Alternatively we could attach version counters to storages and get sharing tracking completely for free

Reply:
Yeah. Along those lines we could just replace Storage with a 1-d Tensor (at least in Python).

Reply:
idk, storages are kind of nice. Can you add a note that says that we don't track `set_`?